nativeasset/contract.go

```diff
@@ -102,8 +104,56 @@ func UnpackNativeAssetCallInput(input []byte) (common.Address, common.Hash, *big
 
 // Run implements StatefulPrecompiledContract
 func (c *NativeAssetCall) Run(accessibleState contract.AccessibleState, caller common.Address, addr common.Address, input []byte, suppliedGas uint64, readOnly bool) (ret []byte, remainingGas uint64, err error) {
-	// input: encodePacked(address 20 bytes, assetID 32 bytes, assetAmount 32 bytes, callData variable length bytes)
-	return accessibleState.NativeAssetCall(caller, input, suppliedGas, c.GasCost, readOnly)
+	if suppliedGas < c.GasCost {
+		return nil, 0, vmerrs.ErrOutOfGas
+	}
+	remainingGas = suppliedGas - c.GasCost
+
+	if readOnly {
+		return nil, remainingGas, vmerrs.ErrExecutionReverted
+	}
+
+	to, assetID, assetAmount, callData, err := UnpackNativeAssetCallInput(input)
+	if err != nil {
+		return nil, remainingGas, vmerrs.ErrExecutionReverted
```

Comment:
It would be helpful to log `err` otherwise it's lost when returning `ErrExecutionReverted`.

Reply:
adding this as a debug log since it could be triggered by p2p input and generally we don't want that type of input to be able to output >=info logs.

I can switch it to info/warn if others feel strongly.